partner_create_by_vat/models/res_partner.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# ©  2015 Forest and Biomass Services Romania
+# See README.rst file on addons root folder for license details
+
+from openerp import models, api, _
+from openerp.exceptions import except_orm
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.one
+    def check_vat_name(self):
+        # If the partner doesn't have the vat field completed, check if the
+        # name field is a valid vat number.
+        if self.name:
+            self.name = self.name.strip().upper()
+            try:
+                from stdnum.eu.vat import validate
+                validate(self.name)
+                self.vat = self.name
+            except:
+                raise except_orm(_('Error!'),
+                                 _("VAT number invalid."))
+        else:
+            raise except_orm(_('Error!'),
+                             _("No VAT number inputted."))
+
+    @api.one
+    def button_get_partner_data(self):
+        if not self.vat:
+            self.check_vat_name()
+        else:
+            self.vat = self.vat.strip().upper()
+        vat_country, vat_number = self._split_vat(self.vat)
+        # Complete country field based on country code
+        self.country_id = self.env['res.country'].search(
+            [('code', 'ilike', vat_country)])[0].id
+        from stdnum.eu.vat import check_vies
```

Comment:
Imports go to head of file. If you don't want it to fail, put it inside a try except block.